composer/datasets/in_context_learning_evaluation.py

```diff
@@ -85,582 +109,764 @@ def _make_padded_input(context_enc, continuation_enc, max_seq_len, pad_tok_id, p
-        with dist.local_rank_zero_download_and_wait(destination_path):
-            if dist.get_local_rank() == 0:
-                get_file(dataset_uri, destination_path, overwrite=True)
-        dataset = load_dataset('json', data_files=destination_path, split='train', streaming=False)
-        self.samples = self._read_dataset(dataset)
-        self.samples = strip_data(self.samples)
+    def __init__(self,
+                 dataset_uri: str,
+                 tokenizer: transformers.PreTrainedTokenizerBase,
+                 max_seq_len: int,
+                 pad_tok_id: int,
+                 num_fewshot: int,
+                 fewshot_random_seed: int,
+                 prompt_string: str,
+                 example_delimiter: str,
+                 continuation_delimiter: str,
+                 destination_path: str,
+                 prelimiter: str = '',
+                 context_key: str = 'context',
+                 answer_key: str = 'answer',
+                 strip_dataset: bool = True,
+                 hf_loading_vars: Dict = None,
+                 hf_parsing_map: Dict = None,
+                 stacked_keys: List[str] = None,
+                 dont_split_keys: List[str] = None,
+                 list_split_keys: List[str] = None,
+                 normal_split_keys: List[str] = None,
+                 ):
+        
         self.tokenizer = tokenizer
+        self.prefix_space = _tokenizer_needs_prefix_space(self.tokenizer)
+
         self.max_seq_len = max_seq_len
         self.pad_tok_id = pad_tok_id
+        self.num_fewshot = num_fewshot
+        # TODO: check this is correct for all dataset types
+        # TODO: change how this is set, using default is unintuitive rn
         self.padding_side = 'left'
-        self.max_answer_length = 0
+        self.prelimiter = prelimiter
+        self.example_delimiter = example_delimiter
+        self.continuation_delimiter = continuation_delimiter
+        self.context_key = context_key
+        self.answer_key = answer_key
+        self.stacked_keys = stacked_keys or ['input_ids', 'labels']
+        self.dont_split_keys = dont_split_keys or []
+        self.list_split_keys = list_split_keys or []
+        self.normal_split_keys = normal_split_keys or []
+
+        hf_loading_vars = hf_loading_vars or {}
+        self.dataset = self._read_dataset(dataset_uri, destination_path, hf_loading_vars, hf_parsing_map)
+        self.strip_data = strip_dataset
+        if self.strip_data:
+            self.dataset = self.dataset.map(strip_data)
+
         fewshot_rng = random.Random(fewshot_random_seed)
-        self.encoded_dataset = self._prep_examples(num_fewshot, prompt_string, example_delimiter,
-                                                   continuation_delimiter, question_prelimiter, fewshot_rng,
-                                                   cot_delimiter)
+        self.dataset = self.dataset.map(
```

Comment:
Just to confirm my understanding of hugging face dataset `map` this is NOT lazy, however it is cached + streamed to ensure memory efficiency, right? Please run two end to end 7b evals on the full gauntlet incl. before/after total run times.

Reply:
yes, the data only exists on disk